rest-service/manager_rest/resource_manager.py

```diff
@@ -479,7 +507,122 @@ def execute_workflow(self,
 
         return new_execution
 
-    def _check_for_any_active_executions(self):
+    @staticmethod
+    def _should_use_system_workflow_executor(execution):
+        """
+        Both system and `regular` workflows are being de-queued, and each kind
+        should be executed using a different executor function
+        (`execute_system_workflow`/`execute_workflow`).
+        * Deployment environment creation and deletion are considered
+        system workflows
+
+        """
+        workflow_id = execution.workflow_id
+        dep_env_workflows = ('create_deployment_environment',
+                             'delete_deployment_environment')
+        if workflow_id in dep_env_workflows or execution.is_system_workflow:
+            return True
+        return False
+
+    def execute_queued_workflow(self, execution):
+        """
+        Whenever an execution is about to change to END status this function
+        is called. Since the execution exists in the DB (it was created and
+        then queued), we don't need to do all the logic of 'execute_workflow',
+        but only to extract the needed information and re-run with the correct
+        workflow executor
+        :param execution: an execution DB object
+        :return:
+        """
+        queue = True
+        force = False
+        deployment = execution.deployment
+        deployment_id = None
+        if deployment:
+            deployment_id = deployment.id
+        workflow_id = execution.workflow_id
+        execution_parameters = execution.parameters
+
+        if self._should_use_system_workflow_executor(execution):
+            # Use `execute_system_workflow`
+            verify_no_executions = True
+            should_queue = False
+            # It means that a system-wide workflow is about to be launched
+            if deployment is None and verify_no_executions:
+                should_queue = self._check_for_any_active_executions(queue)
+
+            # Execution can't currently run,it's queued and will run later
+            if should_queue:
+                return execution
+
+            execution.started_at = utils.get_formatted_timestamp()
+            execution.status = ExecutionState.PENDING
+            self.sm.put(execution)
+            workflow_executor.execute_system_workflow(
+                wf_id=workflow_id,
+                task_id=execution.id,
+                task_mapping=self.task_mapping.get(workflow_id),
+                deployment=deployment,
+                execution_parameters=execution_parameters)
+
+            return execution
+
+        else:  # User `execute_workflow`
+
+            blueprint = self.sm.get(models.Blueprint, deployment.blueprint_id)
+            self._verify_workflow_in_deployment(workflow_id, deployment,
+                                                deployment_id)
+            workflow = deployment.workflows[workflow_id]
+            self._verify_deployment_environment_created_successfully(
```

Comment:
hmm this will throw a RuntimeError if not, aborting the whole `start_queued_executions` loop, perhaps not what we want...

Reply:
and it would mean that the previous execution status update, which called the `start_queued_executions` would return a 500 as well, certainly not what we want ;P